### PR TITLE
Network interface is optional

### PR DESCRIPTION
**Pull request by guillaumerose**
When using vsock, network is not necessary. It simplifies also the routing (no risk of having a default gateway already set, DNS set, etc.)

### CODE REVIEW

File: constants.go [hunk 1]
```diff
@@ -57,9 +54,17 @@ const (
...
 {{- range .ExtraDevices }}
-	{{ . }}
+    {{ . }}
 {{- end }}
   </devices>
 </domain>`
 	VSockDevice = `<vsock model='virtio'><cid auto='yes'/></vsock>`
 )
+
+func NetworkDevice(name string) string {
```
Comment (cfergeau) [inline constants.go]: slight preference for `networkName` here, or just `network` to be consistent with the field name ni `Driver`

File: libvirt.go [hunk 2]
```diff
@@ -304,6 +316,10 @@ func (d *Driver) Start() error {
 		return err
 	}
 
+	if d.Network == "" {
+		return nil
+	}
+
```
Comment (cfergeau) [inline libvirt.go:346]: Hmm this means we have no way of knowing if the VM successfully booted or not? :-/

**Review by cfergeau — COMMENTED** [review]
As usual, can you expand the commit log a bit to explain why we need this change? (in other words, mention the gvisor-vsock-tap work you've been doing which will need this)

**Review by cfergeau — APPROVED** [review]

File: libvirt.go (hunk 2 above)
Comment (guillaumerose) [reply libvirt.go:346]: Yes, sadly. The current method is not particularly better: just polling the IP. We could probably scan the console and wait for the login pompt.

File: constants.go (hunk 1 above)
Comment (guillaumerose) [reply constants.go]: renamed!

**Review by cfergeau — APPROVED** [review]

**Pull request merged**